### PR TITLE
Feature/132625/gallery common panel

### PR DESCRIPTION
**Pull request by nicholasmelwell**
Create Dialog Component.
Create Gallery Controller and Gallery Panel Settings

### CODE REVIEW

File: wp-content/themes/core/pcss/components/_dialog.pcss [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	display: none;
+}
+
+.c-dialog__overlay {
+	background-color: var(--color-goethite);
+	bottom: 0;
+	color: var(--color-white);
+	left: 0;
+	overflow: scroll;
+	position: fixed;
```
Comment (ryanurban) [inline wp-content/themes/core/pcss/components/_dialog.pcss]: For the sake of robusticity (new made up word) I prefer to have a wrapper inside of a parent fixed container that handles overflow scrolling sort of stuff. Gives us a greater degree of dependability on this fixed container may or may not interact in positioning when other site areas come into play.

File: wp-content/themes/core/pcss/components/_dialog.pcss [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	right: 0;
+	top: 0;
+	width: 100%;
+	z-index: 2;
```
Comment (ryanurban) [inline wp-content/themes/core/pcss/components/_dialog.pcss]: Have we done any sort of z-indexery as vars, might be worth spinning something up so we can track our usage a bit (can probably wait for later, making a note).

File: wp-content/themes/core/pcss/components/_dialog.pcss [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	@media (--viewport-medium) {
+		padding-top: 140px;
+		width: 100%;
+	}
+}
+
+.c-dialog__content-inner {
+	margin: 0 auto;
+	padding: 0 20px;
+	max-width: 1000px;
```
Comment (ryanurban) [inline wp-content/themes/core/pcss/components/_dialog.pcss]: Let's either add this to the other grid vars as an available max-width or pick one that exists that is close enough

File: wp-content/themes/core/pcss/components/_dialog.pcss [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+		}
+	}
+}
+
+.c-dialog__header {
+	background-color: var(--color-goethite);
+	border-bottom: 1px solid var(--color-white);
+	margin-bottom: 47px;
+	padding: 20px;
+	position: fixed;
```
Comment (ryanurban) [inline wp-content/themes/core/pcss/components/_dialog.pcss]: See above, going fixed on this in particular is fragile, depending on header title or whatever content may end up in the header from a height perspective. We can remove the fixed handling once you update the fixed item above and for now just allow the whole overlay to scroll, we'll do a refinement pass after this first pass on common panels). We could also potentially try to handle this as a flex situation where the header remains stuck and content fills up whatever area is left if that wouldn't take much time.

File: wp-content/themes/core/pcss/content/panel/_gallery.pcss [hunk 5]
```diff
@@ -6,38 +6,110 @@
...
+		padding-bottom: var(--spacer-2);
+
+		@media (--viewport-medium) {
+			padding-bottom: var(--spacer-4);
+		}
+	}
+}
+
+.site-panel--gallery__content {
+	max-width: 848px;
```
Comment (ryanurban) [inline wp-content/themes/core/pcss/content/panel/_gallery.pcss]: Same notes on these max-widths or do you feel that these are so specific that they aren't worth moving into var form?

File: wp-content/themes/core/pcss/content/panel/_gallery.pcss [hunk 6]
```diff
@@ -6,38 +6,110 @@
...
-		.tribe-image {
-			position: absolute;
-			width: 100%;
-			height: 100%;
-			top: 0;
-			left: 0;
-			background-position: 50% 50%;
-			background-size: cover;
+	.t-content & {
+		margin-top: 20px;
```
Comment (ryanurban) [inline wp-content/themes/core/pcss/content/panel/_gallery.pcss]: This should use an existing spacer var or choose the closest, we shall maintain spacing system integrity!

File: wp-content/themes/core/pcss/content/panel/_gallery.pcss [hunk 7]
```diff
@@ -6,38 +6,110 @@
...
+	}
+}
+
+.site-panel--gallery__slider {
```
Comment (ryanurban) [inline wp-content/themes/core/pcss/content/panel/_gallery.pcss]: I _think_ we should turn a lot of the themery that is part of skinning this slider into a slider theme that can be used anywhere through the use of a c-slider--variation parent class. Let's chat.

File: wp-content/themes/core/pcss/utilities/variables/_colors.pcss [hunk 8]
```diff
@@ -34,6 +34,7 @@
 	--color-nevada: #606a6b;
 	--color-submarine: #b9c2c3;
 	--color-desert-storm: #f2f1f0;
+	--color-goethite: #222222;
```
Comment (ryanurban) [inline wp-content/themes/core/pcss/utilities/variables/_colors.pcss]: Micro optimizations ftw! `#222` 💨

File: wp-content/themes/core/pcss/content/panel/_gallery.pcss (hunk 5 above)
Comment (nicholasmelwell) [reply wp-content/themes/core/pcss/content/panel/_gallery.pcss]: These are very specific to the panels that I've noticed so far. If we do do this, I'd recommend some consistency from design if possible. I'm seeing this mostly within the panel / section header areas with the panel title and description.

Comment (ryanurban) [reply wp-content/themes/core/pcss/content/panel/_gallery.pcss]: That's fair, could you follow up them and see, it's not a bad idea to have a consistent set in place ➕

**Comment by speerface** [comment]
Same deal here - as soon as Ryan signs off, you can go ahead and merge, Nicholas!

**Comment by ryanurban** [comment]
Closing this PR out for now as we'll need to update this following the greater reorg/rearchitecture of SQ1 and moving to blocks : )

**Pull request closed without merging**